src/Compilers/VisualBasic/Portable/VBResources.resx

```diff
@@ -5695,4 +5695,16 @@
   <data name="ERR_InvalidExperimentalDiagID" xml:space="preserve">
     <value>The diagnosticId argument to the 'Experimental' attribute must be a valid identifier</value>
   </data>
+  <data name="WRN_LockTypeUnsupported" xml:space="preserve">
+    <value>A value of type 'System.Threading.Lock' in SyncLock will use likely unintended monitor-based locking. Consider manually calling 'Enter' and 'Exit' methods in a Try/Finally block instead.</value>
```

Comment:
>'Enter' and 'Exit' [](http://example.com/codeflow?start=139&length=18)

Should this be "'EnterLockScope' and 'Dispose' methods" instead?

Reply:
No, the `Lock.Scope` type returned by `EnterLockScope` cannot be used in VB as it's a `ref struct`, so this warning suggests using `Lock.Enter` and `Lock.Exit` instead.